pkg/querier/querier.go

```diff
@@ -119,28 +121,31 @@ func New(cfg Config, distributor Distributor, chunkStore ChunkStore) (storage.Qu
 }
 
 // NewQueryable creates a new Queryable for cortex.
-func NewQueryable(dq, cq storage.Queryable, distributor Distributor, cfg Config) storage.Queryable {
+func NewQueryable(distribQuerier, storeQuerier storage.Queryable, chunkIterFn chunkIteratorFunc, cfg Config) storage.Queryable {
 	return storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
 		q := querier{
-			distributor: distributor,
 			ctx:         ctx,
 			mint:        mint,
 			maxt:        maxt,
+			chunkIterFn: chunkIterFn,
+		}
+
+		dqr, err := distribQuerier.Querier(ctx, mint, maxt)
+		if err != nil {
+			return nil, err
 		}
 
+		q.primary = dqr
```

Comment:
I find a bit unclear naming `primary` the `distributorQuerier`. I would be more explicit it, and just call it `distributorQuerier`.

Reply:
My thinking was that it doesn't have to be distributorQueryable only, but we can also change it back to something else in the future when needed.